### PR TITLE
Deprecate updates field

### PR DESCRIPTION
**Pull request by NikhilGupta36**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: workgroup/src/main/java/software/amazon/athena/workgroup/UpdateHandler.java [hunk 1]
```diff
@@ -92,9 +88,25 @@ private UpdateWorkGroupResponse updateWorkgroup(final ResourceModel model) {
       }
 
       // Handle modifications to WorkGroup configuration
-      return clientProxy.injectCredentialsAndInvokeV2(updateWorkGroupRequest, athenaClient::updateWorkGroup);
+      final UpdateWorkGroupRequest.Builder updateRequestBuilder = UpdateWorkGroupRequest.builder()
+          .workGroup(newModel.getName())
+          .description(newModel.getDescription() != null ? newModel.getDescription() : HandlerUtils.DEFAULT_DESCRIPTION)
+          .state(newModel.getState() != null ? newModel.getState() : HandlerUtils.DEFAULT_STATE);
+
+      // Prioritize looking at WorkGroupConfigurationUpdates field
```
Comment (cmihran) [inline workgroup/src/main/java/software/amazon/athena/workgroup/UpdateHandler.java:96]: This comment is out of date with what the code does (it does opposite)

File: workgroup/src/main/java/software/amazon/athena/workgroup/UpdateHandler.java [hunk 2]
```diff
@@ -92,9 +88,25 @@ private UpdateWorkGroupResponse updateWorkgroup(final ResourceModel model) {
...
+      if (newModel.getWorkGroupConfiguration() != null) {
+        updateRequestBuilder.configurationUpdates(translator.createSdkConfigurationUpdatesFromCfnConfiguration(newModel.getWorkGroupConfiguration()));
+      }else if (newModel.getWorkGroupConfigurationUpdates() != null) {
```
Comment (cmihran) [inline workgroup/src/main/java/software/amazon/athena/workgroup/UpdateHandler.java:99]: Need extra space after the first } in this line

**Review by cmihran — CHANGES_REQUESTED** [review]

File: workgroup/src/main/java/software/amazon/athena/workgroup/UpdateHandler.java (hunk 2 above)
Comment (NikhilGupta36) [reply workgroup/src/main/java/software/amazon/athena/workgroup/UpdateHandler.java]: Done.

**Review by cmihran — APPROVED** [review]

**Comment by dontirun** [comment]
The documentation on the CloudFormation page needs to be updated to remove this field from the "Updating an Athena WorkGroup" examples
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-athena-workgroup.html

**Pull request merged**